packages/bus/src/Brokers/RMQEventBus.spec.ts

```diff
@@ -0,0 +1,560 @@
+// eslint-disable-next-line max-classes-per-file
+import { RMQEventBus } from './RMQEventBus';
+import { RundomString } from '../utils';
+import { ConnectionFactory } from '../Factories/ConnectionFactory';
+import {
+  bind,
+  Command,
+  Configuration,
+  Event,
+  EventBus,
+  EventHandler
+} from '@secbox/core';
+import { injectable } from 'tsyringe';
+import { ConfirmChannel, Connection } from 'amqplib';
+import { anything, instance, mock, reset, verify, when } from 'ts-mockito';
+// Temporary fix for https://github.com/NagRock/ts-mockito/issues/191
+const resolvableInstance = <T extends object>(mockInstance: T): T =>
+  new Proxy<T>(instance(mockInstance), {
+    get(target, prop, receiver) {
+      if (
+        ['Symbol(Symbol.toPrimitive)', 'then', 'catch'].includes(
+          prop.toString()
+        )
+      ) {
+        return undefined;
+      }
+
+      return Reflect.get(target, prop, receiver);
+    }
+  });
+
+describe('RMQEventBus', () => {
+  let config: Configuration;
+  const mockedConnectionFactory = mock<ConnectionFactory<Connection>>();
+  const mockedConnection = mock<Connection>();
+  const mockedConfirmChannel = mock<ConfirmChannel>();
+
+  beforeEach(() => {
+    when(
+      mockedConfirmChannel.consume(anything(), anything(), anything())
+    ).thenResolve({ consumerTag: `agent:${RundomString(22)}` } as any);
+
+    when(
+      mockedConfirmChannel.assertExchange(anything(), anything(), anything())
+    ).thenResolve();
+
+    when(
+      mockedConfirmChannel.assertQueue(anything(), anything())
+    ).thenResolve();
+
+    const confirmChannel = resolvableInstance(mockedConfirmChannel);
+    when(mockedConnection.createConfirmChannel()).thenResolve(
+      confirmChannel as any
+    );
+
+    const connection = resolvableInstance(mockedConnection);
+    when(mockedConnectionFactory.create(anything(), anything())).thenResolve(
+      connection
+    );
+
+    config = new Configuration({
+      cluster: 'amqps://test-url.com:5672',
+      exchange: 'EventBus',
+      clientQueue: 'agent:nnCF9MfHpbvdJVtSbQfKa1',
+      connectTimeout: 10000,
+      heartbeatInterval: 5000,
+      reconnectTime: 5000,
+      credentials: {
+        token: 'weobbz5.nexa.vennegtzr2h7urpxgtksetz2kwppdgj0'
+      }
+    });
+  });
+
+  afterEach(() => {
+    reset(mockedConnectionFactory);
+    reset(mockedConnection);
+    reset(mockedConfirmChannel);
+  });
+
+  describe('init', () => {
+    let eventBus: EventBus;
```

Comment:
Your SUT is `RMQEventBus` and not `EventBus`

```suggestion
    let eventBus!: RMQEventBus;
```